### DISCUSSION TITLE
Unexpected Render with e.g. `useFormState({ control, name: "test.0.lastName" })`

### DISCUSSION THREAD

**Discussion by LydiaF** [react-hook-form/react-hook-form#9738] [Q&A] [answered]
### Version Number

7.37

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-usefieldarray-forked-03k7p5?file=/src/index.tsx:458-509

### Steps to reproduce

Dear Bill/React Hook Team

1. Go to https://codesandbox.io/s/react-hook-form-usefieldarray-forked-03k7p5?file=/src/index.tsx:458-509
2. Click on "Submit"
3. Notice the re-render/console.log

### Expected behaviour

I would not have expected the component to re-render, as the error is in `firstName`, not `lastName`. Is this intended?  If so, would it not be better if it did not rerender? I would like to isolate the re-rendering to just the parts of the form that have errors.

Thanks for looking into this,

Lydia

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
Each submission will trigger a re-render to update submitted and submitCount, which in theory it should not impact your app performance.  I think if your app is suffering performance issues with a single render that doesn't even touch the dom, there is some serious issue you need to check within your app.

**Comment by LydiaF** [comment]
Hey Bill, thank you so much for the quick reply :)

In our actual app we are using trigger, not submit, at this moment. I'm sorry, I should have made the code sandbox more similar to what we are doing. Here is an updated version: https://codesandbox.io/s/react-hook-form-usefieldarray-forked-nfmcqn?file=/src/index.tsx

As you can see, because of the way we are doing the error feedback, each part of the form renders on validation. It just seems a bit overkill and I wanted to look into it.

Thank you so much for everything!!!!

**Comment by bluebill1049** [comment, marked as answer]
Thanks for the updated CSB, calling `trigger()` will validate the entire form, it's pretty much the same as `handleSubmit` for validation.

**Reply by LydiaF** [reply]
Okay, thank you for the information :)
